clients/src/main/java/org/apache/kafka/common/protocol/types/Struct.java

```diff
@@ -278,6 +278,16 @@ public ByteBuffer getBytes(String name) {
         return (ByteBuffer) result;
     }
 
+    public byte[] getByteArray(String name) {
```

Comment:
Why do we need this?

Reply:
The generated code uses `byte[]` to represent fields with Bytes type.  The existing Struct code uses `ByteBuffer`.  That's why this function is needed now, whereas before we just had a function to get BBs.

This is kind of a debatable issue, I guess.  ByteBuffer is not more efficient (in fact, it's less efficient) unless you start sharing memory between multiple buffers, or using off-heap buffers, etc.  I can see a few places in the message processing code where we just turn around immediately and load / unload the BB to an array, which is just pure overhead.

On the other hand, perhaps it would be easier to integrate with the existing code if we provided ByteBuffer accessors and so forth?

The "records" fields are probably where we should get fancy with our optimizations.

Perhaps we should defer this to a follow-up discussion once we get the basic stuff in?

Reply:
Yeah, I was thinking that it would be best to avoid changing things unless they were actually better. Using `byte[]` seems like going in the wrong direction even if in some cases we don't take advantage of using `ByteBuffer`.

Reply:
Perhaps we can revisit this in a follow-on JIRA?  I'm open to changing it, or even making it ByteBuffer everywhere, but for now byte[] is simpler and in some cases more efficient.